docker/bookkeeper/entrypoint.sh

```diff
@@ -48,31 +47,143 @@ export BK_tlsTrustStoreType=JKS
 export BK_tlsTrustStore=/var/private/tls/bookie.truststore.jks
 export BK_tlsTrustStorePasswordPath=/var/private/tls/bookie.truststore.passwd
 
-echo "creating directories for ledger and journal"
-create_dir "${BK_journalDirectories}"
-create_dir "${BK_ledgerDirectories}"
-
-echo "wait for zookeeper"
-until zk-shell --run-once "ls /" ${BK_zkServers}; do sleep 5; done
-
-# We need to update the metadata endpoint and Bookie ID before attempting to delete the cookie
-sed -i "s|.*metadataServiceUri=.*\$|metadataServiceUri=${BK_metadataServiceUri}|" /opt/bookkeeper/conf/bk_server.conf
-if [ ! -z "$BK_useHostNameAsBookieID" ]; then
-  sed -i "s|.*useHostNameAsBookieID=.*\$|useHostNameAsBookieID=${BK_useHostNameAsBookieID}|" ${BK_HOME}/conf/bk_server.conf
-fi
-
-if [ `find $BK_journalDirectory $BK_ledgerDirectories $BK_indexDirectories -type f 2> /dev/null | wc -l` -gt 0 ]; then
-  # The container already contains data in BK directories. This is probably because
-  # the container has been restarted; or, if running on Kubernetes, it has probably been
-  # updated or evacuated without losing its persistent volumes.
-  echo "data available in bookkeeper directories; not formatting the bookie"
-else
-  # The container does not contain any BK data, it is probably a new
-  # bookie. We will format any pre-existent data and metadata before starting
-  # the bookie to avoid potential conflicts.
-  echo "format bookie data and metadata"
-  /opt/bookkeeper/bin/bookkeeper shell bookieformat -nonInteractive -force -deleteCookie
-fi
-
-echo "start bookie"
-/opt/bookkeeper/scripts/entrypoint.sh bookie
+# Create directories for multiple ledgers and journals if specified.
+create_bookie_dirs() {
+  IFS=',' read -ra directories <<< $1
+  for i in "${directories[@]}"
+  do
+      mkdir -p $i
+      if [ "$(id -u)" = '0' ]; then
+          chown -R "${BK_USER}:${BK_USER}" $i
+      fi
+  done
+}
+
+wait_for_zookeeper() {
+    echo "Waiting for zookeeper"
+    until zk-shell --run-once "ls /" ${BK_zkServers}; do sleep 5; done
+}
+
+
+create_zk_root() {
+    if [ "x${BK_CLUSTER_ROOT_PATH}" != "x" ]; then
+        echo "Creating the zk root dir '${BK_CLUSTER_ROOT_PATH}' at '${BK_zkServers}'"
+        zk-shell --run-once "create ${BK_CLUSTER_ROOT_PATH} '' false false true" ${BK_zkServers}
+    fi
+}
+
+configure_bk() {
+    # We need to update the metadata endpoint and Bookie ID before attempting to delete the cookie
```

Comment:
Could you add to the comment to elaborate on what's really happening? Explain what a cookie is and why the metadata service endpoint needs to be updated. Otherwise, it is difficult to understand what it going on.